### ISSUE TITLE
Exporter: Allow export a custom value type other than string

### ISSUE THREAD

**Issue by jasonex7** [primefaces/primefaces#10404]
### Description

Actually with stronglyTypedCells = true, PrimeFaces tries to guess the type of the cell value from a string value. I suggest to allow the user could give the actual value with the appropriate java type in order PrimeFaces could handle more accurately instead of trying to guess. That would allow us to avoid exporting some cells as numeric and some others as text in a single column with Excel  Exporter (see image).

![image](https://github.com/primefaces/primefaces/assets/16229253/b2d98ad5-f3d3-410a-b710-debcd1a0e124)

### Describe the solution you would like

By using already provided ways to set custom export values in <p:column> tag like attributes `exportValue` and `exportFunction` one could set custom objects as values. By this way we could send Number or Date objects directly so PrimeFaces don't have to guess from string values.

Also in the future we could set some custom CellValueExporter (a hypothetical new interface) object in `ExcelOptions` to allow the user could customize `org.apache.poi.ss.usermodel.Cell` value.

### Additional context

[errror_reproducer.zip](https://github.com/primefaces/primefaces/files/12196012/errror_reproducer.zip)

**Comment by jasonex7** [comment]
I'd like to provide a PR, but first I want to confirm with team members if my proposed solution would be OK. What do you think @melloware?

**Comment by melloware** [comment]
I would say submit the PR and we can review!

**Comment by jasonex7** [comment]
With the proposed solution one could use the following xhtml code to export custom objects other than String:

```xhtml
<p:dataTable id="table" var="testObject" value="#{testView.list}">
    <p:column headerText="Name">
        <h:outputText value="#{testObject.name}" />
    </p:column>
    <p:column headerText="Name Custom" exportValue="#{testObject.name}">
        <h:outputText value="#{testObject.name}" />
    </p:column>
    <p:column headerText="Artist">
        <h:outputText value="#{testObject.artist}" />
    </p:column>
    <p:column headerText="Released">
        <h:outputText value="#{testObject.released}" />
    </p:column>
    <p:column headerText="LocalDate" exportValue="#{testObject.localDate}">
        <h:outputText value="#{testObject.localDate}" />
    </p:column>
    <p:column headerText="LocalDateTime" exportValue="#{testObject.localDateTime}">
        <h:outputText value="#{testObject.localDateTime}" />
    </p:column>
    <p:column headerText="LocalTime" exportValue="#{testObject.localTime}">
        <h:outputText value="#{testObject.localTime}" />
    </p:column>
</p:dataTable>
```

Resulting in excel exporting like this

![image](https://github.com/primefaces/primefaces/assets/16229253/1286dbd4-6638-41bb-a6e9-a2d12e17a341)


Also I attach [primefaces-test-dataexporter.zip](https://github.com/primefaces/primefaces/files/12197388/primefaces-test-dataexporter.zip)

**Comment by melloware** [comment]
Nice it looks like your PR has some compilation or test failures!

**Comment by jasonex7** [comment]
Yes, I fixed in the next commit [7890a96](https://github.com/jasonex7/primefaces/commit/7890a96aaf6008dc11bfbcb58ae843f1e8ecf80b) so now the build is OK.
